src/Moryx.AbstractionLayer.Products.Endpoints/ProductManagementController.cs

```diff
@@ -0,0 +1,332 @@
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("producttypes/{id}")]
+        public ActionResult<ProductModel> GetType(long id)
+        {
+            if (id == 0)
+                return BadRequest($"Id was 0");
+            var productType = _productManagement.LoadType(id);
+            if (productType == null)
+                return NotFound();
+            return _productConverter.ConvertProduct(productType, false);
+        }
+
+        [HttpDelete]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("producttypes/{id}")]
+        public ActionResult<bool> DeleteType(long id)
+        {
+            var result = _productManagement.DeleteProduct(id);
+            if (!result)
+                return NotFound();
+            return result;
+        }
+
+        [HttpPut]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Route("producttypes/{id}")]
+        public ActionResult<long> UpdateType(long id, ProductModel modifiedType)
+        {
+            if (modifiedType == null)
+                return BadRequest($"Modified product type was null");
+            var type = _productManagement.LoadType(id);
+            if (type == null)
+                return BadRequest($"No product type with id {modifiedType.Id} was found");
+            type = _productConverter.ConvertProductBack(modifiedType, (ProductType) type);
+            return _productManagement.SaveType(type);
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("producttypes/{identifier}/{revision}")]
+        public ActionResult<ProductModel> GetType(string identifier, short revision)
+        {
+
+            if (identifier == null)
+                return BadRequest($"Identity was null");
+            var identity = new ProductIdentity(WebUtility.HtmlEncode(identifier), revision);
+            var productType = _productManagement.LoadType(identity);
+            if (productType == null)
+                return NotFound();           
+            return _productConverter.ConvertProduct(productType, false);
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("producttypes/{newIdentifier}/{newRevision}")]
+        public ActionResult<ProductModel> Duplicate(string newIdentifier, short newRevision, ProductModel template)
+        {
+            if (template == null)
+                return BadRequest($"Template was null");
+            var identity = new ProductIdentity(WebUtility.HtmlEncode(newIdentifier), newRevision);
```

Comment:
You won't be able to load under the new identity, since it does not exist yet. Encoding it also doesn't seem like a good idea. Either its a number, than nothing happens or it's a weird character sequence that just get's worse.